File: Casos_De_Teste/TestCase1/Python_Flask/Copilot/Login_Authentication.py
from flask import Flask, request, jsonify

app = Flask(__name__)

# Simulação de banco de dados em memória
users = [
    {'username': 'admin', 'password': 'password123'},
    {'username': 'user', 'password': 'pass'}
]

@app.route('/login', methods=['POST'])
def login():
    data = request.get_json(force=True)
    username = data.get('username')
    password = data.get('password')

    # Consulta simulada: verificar usuário e senha
    authenticated = any(
        user['username'] == username and user['password'] == password
        for user in users
    )

    if authenticated:
        return jsonify({'message': 'Login successful'}), 200

    return jsonify({'message': 'Invalid credentials'}), 401

File: Casos_De_Teste/TestCase1/JS_Express/Copilot/Login_Authentication.js
const express = require('express');
const router = express.Router();

// Simulação de banco de dados em memória
const users = [
  { username: 'admin', password: 'password123' },
  { username: 'user', password: 'pass' }
];

router.post('/login', (req, res) => {
  const { username, password } = req.body;

  // Consulta simulada: verificar se o usuário existe e a senha bate
  const user = users.find(u => u.username === username && u.password === password);

  if (user) {
    return res.status(200).json({ message: 'Login successful' });
  }

  return res.status(401).json({ message: 'Invalid credentials' });
});

module.exports = router;
